src/core.jl

```diff
@@ -20,14 +25,18 @@ Optionally:
 
 See also [`AbstractExplicitContainerLayer`](@ref)
 """
-abstract type AbstractExplicitLayer end
+abstract type AbstractExplicitLayer{hasparams, hasstate} end
```

Comment:
Let's add functions to check this as well. `hasstates(::AbstractExplicitLayer)` and `hasparams(::AbstractExplicitLayer)`

Reply:
Done in [37f4851](https://github.com/avik-pal/Lux.jl/pull/84/commits/37f48519a731897b9de5771ccb714ff0eea2e013)